MAC address recalculation

LostFuse opened the pull request:
* While swapping an SD card with the system to another platform the MAC address persisted from the previous one. 
  To prevent that, force MAC recalculation on every boot was enabled.
* Enable overriding environment variables from U-Boot shell for manual MAC
  address modification.

Commit:
bsp: u-boot; disable Vendor Parameter Protection

Signed-off-by: Jakub Lecki <jakub.lecki@3mdeb.com>

Commit:
bsp: u-boot: store generated mac address in an environment variable

For persistency between power cycles

Signed-off-by: Jakub Lecki <jakub.lecki@3mdeb.com>

tomzy-0 commented on recipes-bsp/u-boot/u-boot/0004-sun8i-disable-Vendor-Parameter-Protection.patch:
```diff
@@ -0,0 +1,28 @@
+  * A23 specific configuration
+  */
+ 
++#define CONFIG_ENV_OVERWRITE 1
```
@leckijakub A short comment explaining why we are setting this config would be nice

LostFuse replied:
@TomaszAIR I have added a comment

tomzy-0 commented on recipes-bsp/u-boot/u-boot/0005-generate-random-ethaddr-by-hardcoding-invalid-addres.patch:
```diff
@@ -0,0 +1,37 @@
+ 		net_random_ethaddr(pdata->enetaddr);
+ 		printf("\nWarning: %s (eth%d) using random MAC address - %pM\n",
+ 		       dev->name, dev->seq, pdata->enetaddr);
++		eth_env_set_enetaddr_by_index("eth", dev->seq, pdata->enetaddr);
```
@leckijakub Let's add some comments also here, maybe a small printf informing that ethaddr was saved in the environment variable

LostFuse replied:
@TomaszAIR comment along with a `printf` added

Commit:
bsp: u-boot: force recalculating the MAC address on every boot

This is a walkaround for the problem of persisting MAC address when
running the system on multiple boards.

Signed-off-by: Jakub Lecki <jakub.lecki@3mdeb.com>
